### PR TITLE
Session page

### LINKED ISSUE #257: Implement session page
Design: https://github.com/GoogleChrome/devsummit/issues/217

### PR DESCRIPTION
**Pull request by surma**
Closes #257 

@argyleink I took some liberties in the design so this will definitely profit from a review from you :D

### CODE REVIEW

File: src/_includes/add-to-cal/index.njk [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{%- set css = "/_includes/add-to-cal/style.css" -%}
+{% css page,
+css %}
+<div class="{% className css, 'root' %}">
+  <span class="{% className css, 'icon' %}">
+    Add to:
+  </span>
+  <span class="{% className css, 'links' %}">
+    <a href="#">Google</a>
```
Comment (surma) [inline src/_includes/add-to-cal/index.njk]: I haven’t quite figured out how to populate the links here. I will do that in a separate PR

**Comment by surma** [comment]
https://deploy-preview-263--cds2019.netlify.com/devsummit/sessions/making-the-web-faster/

File: .eleventy.js [hunk 2]
```diff
@@ -93,6 +103,9 @@ module.exports = function(eleventyConfig) {
 
   /** Format a date in the timezone of the conference */
   eleventyConfig.addShortcode('confDate', (timestamp, format) => {
+    if (typeof timestamp === 'string') {
+      timestamp = new Date(timestamp);
+    }
```
Comment (jakearchibald) [inline .eleventy.js]: Is this safe to do? If a value like `'2019/11/11 09:00'` is passed, the date will depend on the timezone of the JavaScript environment.

File: src/_includes/layouts/session/index.njk [hunk 3]
```diff
@@ -0,0 +1,50 @@
+{% extends 'layouts/main/index.njk' %}
+
+{% block content %}
+  {%- set moduleCSS = "/_includes/module.css" -%}
+  {% css page,
+  moduleCSS %}
```
Comment (jakearchibald) [inline src/_includes/layouts/session/index.njk]: Out of curiosity: why the line break?

File: src/_includes/layouts/session/index.njk [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  {%- set css = "/_includes/content.css" -%}
+  {% css page,
+  css %}
+  <div class="{% className moduleCSS, 'root-alt' %} {% className sessionCSS, 'root' %}">
+    <section class="{% className sessionCSS, 'factsheet' %}">
+      <h1 class="{% className moduleCSS, 'heading' %} {% className sessionCSS, 'heading' %}">{{title}}</h1>
+      <ul class="{% className sessionCSS, 'taglist' %}">
+        {% for tag in session_tags -%}
+          <li class="{% className sessionCSS, 'tag' %}">
+            <a href="#">{{tag}}</a>
```
Comment (jakearchibald) [inline src/_includes/layouts/session/index.njk]: Do we know where these will point for the next launch?

File: src/_includes/calendar-widget/index.njk [hunk 5]
```diff
@@ -0,0 +1,14 @@
+{%- set css = "/_includes/calendar-widget/style.css" -%}
+{% css page, css %}
+<aside class="{% className css, 'root' %}">
```
Comment (jakearchibald) [inline src/_includes/calendar-widget/index.njk]: This uses the conference start date, but I think the intent is for it to be the session time.

File: src/_includes/calendar-widget/index.njk [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  <div class="{% className css, 'icon' %}">
+    <span class="{% className css, 'month' %}">{% confDate start, 'MMM' %}</span>
+    <span class="{% className css, 'day' %}">{% confDate start, 'DD' %}</span>
+  </div>
+  <div class="{% className css, 'date-container' %}">
+    <span class="{% className css, 'weekday' %}">{% confDate start, 'dddd' %},</span>
+    <span class="{% className css, 'full-date' %}">{% confDate start, 'MMMM DD' %}</span>
+  </div>
+  <div class="{% className css, 'time' %}">{% confDate start, 'h:mm A' %}</div>
```
Comment (jakearchibald) [inline src/_includes/calendar-widget/index.njk]: In the design, the minutes are only displayed if it isn't on the hour.

**Comment by surma** [comment]
Waiting for @argyleink to press that green review button ⏳

**Comment by surma** [comment]
Boom

**Pull request merged**